packages/front-end/components/Forms/CodeTextArea.tsx

```diff
@@ -49,6 +53,28 @@ export default function CodeTextArea({
     otherProps.error = semicolonWarning;
   }
 
+  const theme = useMemo(() => {
+    let actualTheme = LIGHT_THEME;
+
+    try {
+      const fromStorage = localStorage.getItem("gb_ui_theme");
+      if (
+        !fromStorage &&
+        window.matchMedia("(prefers-color-scheme: dark)")?.matches
+      ) {
+        actualTheme = DARK_THEME;
+      }
+
+      if (fromStorage === "dark") {
+        actualTheme = DARK_THEME;
+      }
+    } catch (e) {
+      return actualTheme;
+    }
+
+    return actualTheme;
+  }, []);
```

Comment:
When we need to do something like this again, it would be a good time to do a refactor to add it as a context provider and expose a `useAppearanceTheme()` hook or similar.